classes/responsetype/single.php

```diff
@@ -109,16 +109,15 @@ public function insert_response($responsedata) {
             foreach ($response->answers[$this->question->id] as $answer) {
                 if (isset($this->question->choices[$answer->choiceid])) {
                     if ($this->question->choices[$answer->choiceid]->is_other_choice()) {
-                        // If no input specified, ignore this choice.
-                        if (empty($answer->value) || preg_match("/^[\s]*$/", $answer->value)) {
-                            continue;
+                        // If no input specified, don't save other choice, but always save base response.
+                        if (!empty($answer->value) && !preg_match("/^[\s]*$/", $answer->value)) {
+                            $record = new \stdClass();
+                            $record->response_id = $response->id;
+                            $record->question_id = $this->question->id;
+                            $record->choice_id = $answer->choiceid;
+                            $record->response = clean_text($answer->value);
+                            $DB->insert_record('questionnaire_response_other', $record);
                         }
```

Comment:
I'm not 100% sure this is what is desired. If the user didn't select a response, why would a response be saved? Please provide a test case